packages/drivers/odsp-driver-definitions/src/errors.ts

```diff
@@ -42,6 +42,11 @@ export enum OdspErrorType {
     // This condition will result in any local changes being lost (i.e. only way to save state is by user
     // copying it over manually)
     cannotCatchUp = "cannotCatchUp",
+
+    // SPO can occasionally return 403 for r/w operations on document when there is a fail over to another data center.
```

Comment:
Do we have feedback for ODSP on how they're handling these scenarios? Sometimes I feel like there are a lot of problems packed underneath 403. Is this even the right status code from the service? At a glance I would have expected the service to 405. We should consider proposing this feedback to ODSP whenever we come across it, in addition to adapting.

Reply:
Currently they return 403 code with this response: { "code": "accessDenied", "innerError": { "code": "serviceReadOnly" }, "message": "Database Is Read Only" }
Current suggestion is to read the innerMostCode. For now we need to go with this and then I will talk with them about returning another status code but maybe they have this code for such scenarios for vroom apis.